demisto_sdk/commands/validate/validators/BA_validators/BA116_cli_name_should_equal_id.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from typing import Iterable, List, Union
+
+from demisto_sdk.commands.common.constants import GitStatuses
+from demisto_sdk.commands.content_graph.objects.incident_field import IncidentField
+from demisto_sdk.commands.content_graph.objects.indicator_field import IndicatorField
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    FixResult,
+    ValidationResult,
+)
+
+ContentTypes = Union[IncidentField, IndicatorField]
+
+
+class CliNameMatchIdValidator(BaseValidator[ContentTypes]):
+    error_code = "BA116"
+    description = (
+        "validate that the CLI name and the id match for incident and indicators field"
+    )
+    error_message = "The cli name {0} doesn't match the object id {1}"
+    fix_message = "Changing name to be equal to id ({0})."
+    related_field = "cli_name, id"
+    is_auto_fixable = True
+    expected_git_statuses = [GitStatuses.ADDED]
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(
+                    content_item.cli_name, content_item.object_id
+                ),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if content_item.object_id != content_item.cli_name
+        ]
+
+    def fix(self, content_item: ContentTypes) -> FixResult:
```

Comment:
why does is_valid get a list, but fix gets a single item? Is that something we can fix in the infra? Seems very counter intuitive

Reply:
is_valid gets a list because some validations can be done on the whole list (for example graph validations need 1 execution with the received list)
however, each item needs a fix on its own..
We could change that so that fix gets a list as well but that actually makes it harder in my opinion as it can't be done in list comp so users will need to:
1. create a results list,
2. iterate on all items
3. for each item fix and append the fix result to the list
4. return the list
Rather than fix, and return the fix result.